### PR TITLE
[AKS] `az aks upgrade`: Support tier switch with AKS upgrade

### PR DESCRIPTION
**Pull request by wenxuan0923**
**Related command**

`az aks upgrade`

**Description**

Adding options `--tier` and `--k8s-support-plan` to `az aks upgrade` command to support enable/disable AKSLongTermSupport with upgrade operation.

For example, if a cluster is currently on version 1.26, they can upgrade to 1.27 LTS with command: `az aks upgrade -g myResourceGroup -n myCluster -k 1.27 --tier Premium --k8s-support-plan AKSLongTermSupport`

Similarly, to exit LTS for clusters on 1.27, they can do: `az aks upgrade -g myResourceGroup -n myCluster -k 1.28 --tier Standard --k8s-support-plan KubernetesOfficial`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -859,6 +861,12 @@ def aks_upgrade(cmd,
         mc = client.get(resource_group_name, name)
         return _remove_nulls([mc])[0]
 
+    if tier is not None:
+        instance.sku.tier = tier
+
+    if k8s_support_plan is not None:
+        instance.support_plan = k8s_support_plan
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:868]: Is it necessary to add validation about lts tier requires premium sku?

**Review by FumingZhang — COMMENTED** [review]
lgtm
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98878757&view=results) to validate the change.
- test_aks_upgrade_with_tier_switch

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (wenxuan0923) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:870]: @FumingZhang thanks for reviewing! Yes, we can add it here or rely on the validation on the server side. I just updated the code to raise CLIError for it on L872.

Comment (wenxuan0923) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:870]: New live test passed: https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98938062&view=results

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by FumingZhang** [comment]
PR #29436 just merged, which updated the default API version used by the command group, you'll need to update the recording file to pass the CI. Please also rebase from dev branch to pick up the change.

**Comment by wenxuan0923** [comment]
PR rebased to the latest dev branch

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (sdrinovsky) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:870]: I am having an issue on L872. shouldn't it be instance.sku.tier not instance.tier?

Comment (wenxuan0923) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:870]: @sdrinovsky thanks for bringing this to my attention, I started a PR to get it fixed https://github.com/Azure/azure-cli/pull/29949

**Pull request merged**